File: backend/src/models/Delivery.js
const mongoose = require('mongoose');

const deliverySchema = new mongoose.Schema(
  {
    subscription: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Subscription',
      default: null,
    },
    order: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Order',
      default: null,
    },
    customer: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    farmer: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    product: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Product',
      required: true,
    },
    date: { type: String, required: true }, // YYYY-MM-DD
    deliverySlot: {
      type: String,
      enum: ['MORNING', 'EVENING', 'BOTH'],
      default: 'MORNING',
    },
    requestedQuantity: { type: Number, required: true },
    deliveredQuantity: { type: Number, required: true, default: 0 },
    unit: { type: String, required: true, default: 'L' },
    pricePerUnit: { type: Number, required: true },
    totalCost: { type: Number, required: true }, // deliveredQuantity * pricePerUnit
    status: {
      type: String,
      enum: ['PENDING', 'DELIVERED', 'SKIPPED', 'CANCELLED', 'FAILED'],
      default: 'PENDING',
    },
  },
  { timestamps: true }
);

deliverySchema.index({ customer: 1, product: 1, date: 1 }, { unique: true });

module.exports = mongoose.model('Delivery', deliverySchema);

File: backend-dotnet/Models/Delivery.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace DailyMarts.Api.Models;

[BsonIgnoreExtraElements]
public class Delivery
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    [BsonElement("subscription")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Subscription { get; set; }

    [BsonElement("order")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Order { get; set; }

    [BsonElement("customer")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Customer { get; set; } = string.Empty;

    [BsonElement("farmer")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Farmer { get; set; } = string.Empty;

    [BsonElement("product")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Product { get; set; } = string.Empty;

    [BsonElement("date")]
    public string Date { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-dd");

    [BsonElement("deliverySlot")]
    public string DeliverySlot { get; set; } = "MORNING"; // MORNING, EVENING, BOTH

    [BsonElement("requestedQuantity")]
    public double RequestedQuantity { get; set; }

    [BsonElement("deliveredQuantity")]
    public double DeliveredQuantity { get; set; } = 0;

    [BsonElement("unit")]
    public string Unit { get; set; } = "L";

    [BsonElement("pricePerUnit")]
    public decimal PricePerUnit { get; set; }

    [BsonElement("totalCost")]
    public decimal TotalCost { get; set; }

    [BsonElement("status")]
    public string Status { get; set; } = "PENDING"; // PENDING, DELIVERED, SKIPPED, CANCELLED, FAILED

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    [BsonIgnore]
    public Product? ProductDetails { get; set; }

    [BsonIgnore]
    public User? CustomerDetails { get; set; }

    [BsonIgnore]
    public User? FarmerDetails { get; set; }
}
